lib/Doctrine/Common/Annotations/CachedReader.php

```diff
@@ -51,16 +53,24 @@ final class CachedReader implements Reader
     private $loadedAnnotations = [];
 
     /**
-     * Constructor.
-     *
-     * @param Reader $reader
-     * @param Cache  $cache
-     * @param bool   $debug
+     * @param Cache|CacheInterface $cache
+     * @param bool                 $debug
      */
-    public function __construct(Reader $reader, Cache $cache, $debug = false)
+    public function __construct(Reader $reader, $cache, $debug = false)
     {
+        if ($cache instanceof Cache) {
+            @trigger_error(sprintf(
+                'Passing a %s instance to %s is deprecated since doctrine/annotations 1.9 and will not be possible anymore in 2.0. Please pass a %s instance',
+                Cache::class,
+                __METHOD__,
+                CacheInterface::class
+            ), E_USER_DEPRECATED);
```

Comment:
Do we have any way to test deprecations currently? I know the phpunit bridge was used on DBAL, and then removed…

Reply:
Using an error handler as we did on the ORM is the simplest way I can think of.